swap/swap.go

```diff
@@ -299,153 +262,34 @@ func (s *Swap) processAndVerifyCheque(cheque *Cheque, p *Peer) (uint64, error) {
-		err = s.store.Get(balanceKey(peer), &peerBalance)
-		s.setBalance(peer, peerBalance)
-	}
-	return
-}
-
-// sendCheque sends a cheque to peer
-// To be called with mutex already held
-// Caller must be careful that the same resources aren't concurrently read and written by multiple routines
-func (s *Swap) sendCheque(swapPeer *Peer) error {
-	peer := swapPeer.ID()
-	cheque, err := s.createCheque(swapPeer)
-	if err != nil {
-		return fmt.Errorf("error while creating cheque: %s", err.Error())
-	}
-
-	log.Info("sending cheque", "honey", cheque.Honey, "cumulativePayout", cheque.ChequeParams.CumulativePayout, "beneficiary", cheque.Beneficiary, "contract", cheque.Contract)
-	s.setCheque(peer, cheque)
-
-	err = s.store.Put(sentChequeKey(peer), &cheque)
-	if err != nil {
-		return fmt.Errorf("error while storing the last cheque: %s", err.Error())
-	}
-
-	emit := &EmitChequeMsg{
-		Cheque: cheque,
-	}
-
-	// reset balance;
-	err = s.resetBalance(peer, int64(cheque.Honey))
-	if err != nil {
-		return err
-	}
-
-	return swapPeer.Send(context.Background(), emit)
-}
-
-// createCheque creates a new cheque whose beneficiary will be the peer and
-// whose amount is based on the last cheque and current balance for this peer
-// The cheque will be signed and point to the issuer's contract
-// To be called with mutex already held
-// Caller must be careful that the same resources aren't concurrently read and written by multiple routines
-func (s *Swap) createCheque(swapPeer *Peer) (*Cheque, error) {
-	var cheque *Cheque
-	var err error
-
-	peer := swapPeer.ID()
-	beneficiary := swapPeer.beneficiary
-
-	peerBalance, exists := s.getBalance(peer)
-	if !exists {
-		return nil, fmt.Errorf("peer not found %v: ", peer)
-	}
-	// the balance should be negative here, we take the absolute value:
-	honey := uint64(-peerBalance)
-
-	var amount uint64
-	amount, err = s.oracle.GetPrice(honey)
-	if err != nil {
-		return nil, fmt.Errorf("error getting price from oracle: %s", err.Error())
-	}
-
-	// if there is no existing cheque when loading from the store, it means it's the first interaction
-	// this is a valid scenario
-	total, err := s.getLastChequeValues(peer)
-	if err != nil && err != state.ErrNotFound {
-		return nil, err
-	}
-
-	cheque = &Cheque{
-		ChequeParams: ChequeParams{
-			CumulativePayout: total + amount,
-			Contract:         s.owner.Contract,
-			Beneficiary:      beneficiary,
-		},
-		Honey: honey,
-	}
-	cheque.Signature, err = cheque.Sign(s.owner.privateKey)
-
-	return cheque, err
-}
-
-func (s *Swap) getLastChequeValues(peer enode.ID) (total uint64, err error) {
-	err = s.loadLastSentCheque(peer)
-	if err != nil {
-		return
-	}
-	lastCheque, exists := s.getCheque(peer)
-	if exists {
-		total = lastCheque.CumulativePayout
-	}
-	return
-}
-
 // Balance returns the balance for a given peer
 func (s *Swap) Balance(peer enode.ID) (int64, error) {
-	var err error
-	peerBalance, ok := s.getBalance(peer)
+	swapPeer, ok := s.peers[peer]
```

Comment:
We should probably use `getPeer` here which is (read-)locked by `peersLock`

Reply:
done